.github/workflows/haskell-ci.yml

```diff
@@ -20,12 +20,15 @@ on:
       - master
 jobs:
   linux:
-    name: Haskell-CI Linux
+    name: Haskell-CI Linux - GHC ${{ matrix.ghc }}
     runs-on: ubuntu-18.04
     container:
       image: buildpack-deps:bionic
+    continue-on-error: ${{ matrix.continue-on-error }}
     strategy:
       matrix:
+        continue-on-error:
+          - false
```

Comment:
the code implies `false` case shouldn't be generated?

Reply:
Right, the presentation that GitHub Actions uses for this is a bit confusing. Supposing we have this:

```yaml
matrix:
  continue-on-error:
    - false
  include:
    - ghc: 8.10.1
    - ghc: 8.8.3
      continue-on-error: true
```

Then the default `continue-on-error` setting will be `false`, and that setting will be inherited for each entry in `include` unless they explicitly override `continue-on-error` (e.g., the `ghc: 8.8.3` entry). If you wanted to be more explicit, you could also do this:

```yaml
matrix:
  include:
    - ghc: 8.10.1
      continue-on-error: false
    - ghc: 8.8.3
      continue-on-error: true
```

I wanted to avoid repeating the line `continue-on-error: false` over and over again, so I didn't opt for this approach. But if you'd prefer that presentation, I could just as well switch to that.

Reply:
Don't you need 

```yaml
matrix:
  continue-on-error: ${{ matrix.continue-on-error }}
  include:
    - ghc: 8.10.1
      continue-on-error: false
    - ghc: 8.8.3
      continue-on-error: true
```

I was quite sure that includes aren't automatically incorporated. Or is `continue-on-error` one of rare keys which are?

Reply:
If you specify `continue-on-error` for each `include` entry, then you don't need a `continue-on-error` entry under `matrix` itself. See [this commit](https://github.com/RyanGlScott/gha-sandbox/commit/73448f81a04679586f3e661efeded375c355e835), as well as its [associated workflow](https://github.com/RyanGlScott/gha-sandbox/actions/runs/426417388).

Note that when you reference `${{ matrix.continue-on-error }}`, you're referencing the value of `continue-on-error` for the particular `include` entry that some job in a matrix is using. As a result, it doesn't really make sense to reference `${{ matrix.continue-on-error }}` inside of the matrix itself.

Reply:
Oh, yes sorry. So there is `${{ matrix.continue-on-error }}`` but it's one level up.

```yaml
continue-on-error: ${{ matrix.continue-on-error }}
matrix:
  include:
    - ghc: 8.10.1
      continue-on-error: false
    - ghc: 8.8.3
      continue-on-error: true
```

OK. I think I understand how things compose here.

And as `continue-on-error` is something I don't expect to be used much, your way of not having it per each `include` is better.